classes/log/SubmissionEmailLogEntry.php

```diff
@@ -18,6 +18,18 @@
 
 namespace PKP\log;
 
+use APP\core\Application;
+use APP\submission\Submission;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Support\Facades\Mail;
+use PKP\core\Core;
+use PKP\db\DAOResultFactory;
+use PKP\facades\Locale;
+use PKP\facades\Repo;
+use PKP\mail\Mailable;
+use PKP\user\User;
+
 class SubmissionEmailLogEntry extends EmailLogEntry
```

Comment:
Based on the current structure, a single model could work. I'm not entirely sure what the original vision behind having separate classes like `SubmissionEmailLog` or `MonographFileEmailLog` was. An advantage I see is that each class maintains its own `assocType` (`assoc_type` in the database table) value. This helps when performing queries, as we can get the value of the assoc type directly from the class. However, passing that value to the methods performing the queries (when using a single class) doesn't seem like it would be a major issue.

Reply:
Yeah, and we can transform classes that hold app specific log-related constants into enums and use just for it. @asmecher knows historical reasons of keeping separate email log classes better, what do you think about their unification into a single model?

Reply:
I'd support reconciling things into a single class. The abstraction isn't worth defending if it's going to make things awkward/verbose with Eloquent.

Reply:
I'll proceed with refactoring the code to use a single class, as it simplifies some things.

Here’s an example of how I was managing class-specific values:

Initialize the value in the constructor of the child class:
https://github.com/pkp/pkp-lib/blob/11f2069f6fa6c08192891d80a01bb66cdb936e17/classes/log/SubmissionEmailLogEntry.php#L88-L91

Add the methods that used such value as a static method on the parent class:
https://github.com/pkp/pkp-lib/blob/11f2069f6fa6c08192891d80a01bb66cdb936e17/classes/log/EmailLogEntry.php#L252-L255
Static method accesses the value like ```
$instance->assocType```

Use method via child class like ```SubmissionEmailLogEntry::getByEventType(...args).```